subprojects/reporting/src/main/java/org/gradle/api/reporting/ConfigurableReport.java

```diff
@@ -48,18 +48,6 @@
     /**
      * Sets the destination for the report.
      *
-     * The file parameter is evaluated as per {@link org.gradle.api.Project#file(Object)}.
-     *
-     * @param file The destination for the report.
-     * @see #getDestination()
-     * @deprecated Replaced by {@link #setDestination(java.io.File)} and {@link #setDestination(org.gradle.api.provider.Provider)}.
-     */
-    @Deprecated
-    void setDestination(Object file);
```

Comment:
We should think about removing the overridden method with the same signature on

- `SimpleReport`
- `TaskGeneratedSingleDirectoryReport`
- `TaskGeneratedSingleFileReport`

These classes are internal but we have an opportunity to get rid of the method completely. Have a look at the uses of the method in other context and see if it makes sense.